skimage/feature/peak.py

```diff
@@ -14,51 +14,119 @@ def _get_high_intensity_peaks(image, mask, num_peaks):
     intensities = image[coord]
     # Highest peak first
     idx_maxsort = np.argsort(-intensities)
-    coord = np.stack(coord, axis=-1)[idx_maxsort]
-    # select num_peaks peaks
+    coord = np.transpose(coord)[idx_maxsort]
+
+    if len(coord):
+        # Use KDtree to find the peaks that are too close to each other
+        tree = spatial.cKDTree(coord)
+
+        indices = tree.query_ball_point(coord, r=min_distance, p=p_norm)
+        rejected_peaks_indices = set()
+        for idx, candidates in enumerate(indices):
+            if idx not in rejected_peaks_indices:
+                candidates.remove(idx)
+                rejected_peaks_indices.update(candidates)
+
+        # Remove the peaks that are too close to each other
+        coord = np.delete(coord, tuple(rejected_peaks_indices), axis=0)
```

Comment:
Several suggestions here: 

- We can skip this step if `min_distance` is 1, correct? 
- Also, it's probably best to update the description of that parameter in the docstring similar to #4218? I think the part with "2 * min_distance + 1" could be clearer.
- As already suggested in #4218 it's probably best to factor this out? Maybe even to Cython to get rid of the slow loop (doesn't have to be this PR)?

Reply:
Thank you @lagru for your review, and sorry for my late answer, I have been too much busy these last few days :pray:.

> - We can skip this step if min_distance is 1, correct?

No, we also want to remove the adjacent peaks (I may be didn't get your point here :confused:).

>  - Also, it's probably best to update the description of that parameter in the docstring similar to #4218? I think the part with "2 * min_distance + 1" could be clearer.

:+1:

 > - As already suggested in #4218 it's probably best to factor this out? Maybe even to Cython to get rid of the slow loop (doesn't have to be this PR)?

Why not, I remember that we already mentioned this ;)

Reply:
@lagru, I factored out this part as you suggested in the `skimage._shared.coord.ensure_spacing` function, I hope you agree with the function name and its API.

Reply:
Thank you. The name seems fine to me. Did you intentionally not replace the appropriate code snippet in [corner_peaks](https://github.com/scikit-image/scikit-image/blob/580629f901fbc2d1c80d80fe15825ee9c12345f0/skimage/feature/corner.py#L1083-L1097) with `ensure_spacing`?

Reply:
With this Fix of `peak_local_max`, `corner_peaks` becomes obsolete and can directly be replaced by the former.